lab_4_summarization_textrank/main.py

```diff
@@ -2,15 +2,33 @@
 Lab 4
 Summarize text using TextRank algorithm
 """
-from typing import Union
+from typing import Union, Any, Type
 
+import re
 from lab_3_keywords_textrank.main import TextEncoder, \
-    TextPreprocessor
+    TextPreprocessor, TFIDFAdapter
+
 
 PreprocessedSentence = tuple[str, ...]
 EncodedSentence = tuple[int, ...]
 
 
+def check_types(variable: Any, possible_var_type: Type, container_value_type: Union[Type, None] = None) -> None:
+    """
+    Checks if the variable is of an apropriate type
+    param: variable
+    param: possible_var_type
+    param: container_value_type (default = None)
+    return:
+    """
+    if not isinstance(variable, possible_var_type) or isinstance(variable, bool):
+        raise ValueError
+    if container_value_type:
```

Comment:
if not -> return

Reply:
Why? It's an optional variable and those actions pass only if i have it

Reply:
I changed it, is this what you meant?